### DISCUSSION TITLE
How to set the locale within Targets

### DISCUSSION THREAD

**Discussion by DrMattG** [ropensci/targets#1008] [Help] [answered]
I have a targets pipeline that needs to import data that is written in Norwegian. My system is set to Norwegian Bokmål and can read the file in outside of the targets. 

The function is as follows: 
```
get_data<-function(file){
  data_list <- rio::import_list(file)
}
```

and the _targets.R is:
```
library(targets)
source("R/functions.R")
list(
  tar_target(file, "data/Overvåking_DATA.xlxs"),
  tar_target(data, get_data(file))
)
```
When I run tar_make() I get the error that `path` does not exist: 'data/OvervÃ¥king_DATA.xlsx'

---
I know this has something to do with the encoding I just dont know where I need to make this explicit in the targets file. Any help would be gratefully recieved!

**Comment by JamesHWade** [comment, marked as answer]
You might be able to use `withr::with_locale()`.

```
get_data<-function(file){
  data_list <- withr::with_locale(<define locale>, rio::import_list(file))
}
```

You can see the `with_locale()` documentation [here](https://withr.r-lib.org/reference/with_locale.html).

**Reply by DrMattG** [reply]
Brilliant thank you!
